### PR TITLE
Switch to yarnpkg client, upgrade typings & dev dependencies

### PR DESCRIPTION
**Pull request by adidahiya**
~~(will add docs for dev environment changes after build passes)~~ done

### CODE REVIEW

File: packages/core/src/components/slider/coreSlider.tsx [hunk 1]
```diff
@@ -153,10 +153,14 @@ export abstract class CoreSlider<P extends ICoreSliderProps> extends AbstractCom
...
-        const { max, min, labelStepSize } = this.props;
+        // explicit typedefs are required because tsc (rightly) assumes that props might be overriden with different
+        // types in subclasses
+        const max: number = this.props.max;
+        const min: number = this.props.min;
+        const labelStepSize: number = this.props.labelStepSize;
         if (this.props.renderLabel === false) { return undefined; }
 
-        const stepSize = Math.round(this.state.tickSize * labelStepSize);
+        const stepSize = Math.round(this.state.tickSize * (labelStepSize as number));
```
Comment (adidahiya) [inline packages/core/src/components/slider/coreSlider.tsx]: unnecessary type assertion here

File: circle.yml [hunk 2]
```diff
@@ -16,27 +16,37 @@ general:
...
+    - ~/blueprint/packages/site-docs/node_modules
+    - ~/blueprint/packages/site-landing/node_modules
+    - ~/blueprint/packages/table/node_modules
   # non-zero exit codes in `dependencies` group will fail the build early
   # so these following commands will block the build and prevent tests
-  post:
-    - lerna bootstrap
-    - npm run build:gulp
+  override:
+    - ./scripts/yarn.sh $YARN_VERSION
```
Comment (rcchen) [inline circle.yml]: Just use `apt-get install yarn` instead of `yarn.sh`

File: package.json [hunk 3]
```diff
@@ -16,92 +16,92 @@
...
+    "source-map-loader": "^0.2.1",
+    "string-replace-loader": "^1.2.0",
+    "stylelint": "^7.11.1",
+    "stylelint-config-palantir": "^2.0.2",
+    "stylelint-scss": "^1.4.4",
+    "ts-loader": "^2.2.0",
+    "ts-quick-docs": "^0.5.3",
+    "tslint": "^4.5.1",
+    "tslint-react": "^3.0.0",
+    "typescript": "^2.3.4",
```
Comment (rcchen) [inline package.json]: Watch out with this!

File: package.json [hunk 4]
```diff
@@ -16,92 +16,92 @@
...
+    "vinyl-source-stream": "^1.1.0",
+    "webpack": "^1.13.2"
```
Comment (rcchen) [inline package.json]: Webpack 3 might be the next thing to get on the train for

File: packages/datetime/package.json [hunk 5]
```diff
@@ -14,11 +14,11 @@
...
-    "bourbon": "4.2.6",
-    "react": "15.5.1",
-    "react-addons-css-transition-group": "15.5.1",
-    "react-dom": "15.5.1",
-    "react-test-renderer": "15.5.4"
+    "bourbon": "^4.3.4",
+    "react": "^15.6.1",
+    "react-addons-css-transition-group": "^15.6.0",
+    "react-dom": "^15.6.1",
+    "react-test-renderer": "^15.6.1"
```
Comment (giladgray) [inline packages/datetime/package.json]: i thought we were in the habit of locking down dev deps? (no `^`?)

Comment (adidahiya) [inline packages/datetime/package.json]: The lockfile locks them down. Keeping the caret lets us run yarn upgrade.

File: package.json (hunk 4 above)
Comment (adidahiya) [inline package.json]: Will do that separately, we're not even on v2 yet

**Comment by adidahiya** [comment]
Closing for now, will reopen when I am able to get back to this in a week or two.

File: .gitignore [hunk 6]
```diff
@@ -12,11 +12,7 @@ node_modules
 npm-debug.log
 lerna-debug.log
 
-# Gradle
-.gradle/
-.node/
-
-# Sass
+# SASS
```
Comment (llorca) [inline .gitignore]: Sass was correct: http://sass-lang.com

Comment (giladgray) [inline .gitignore]: i know, i fixed it already

**Comment by giladgray** [comment]
current status is fucking obnoxious: typescript 2.1 gets installed by `ts-quick-docs` and gets used by `gulp tsc` (instead of the declared dep on ts 2.2.), which does not yet support `object` keyword, so compiler fails of course.

i can't simply update ts-q-d because the compiler API changed significanly. major fudgins.

**Comment by adidahiya** [comment]
Hey I think the `$(npm bin)/tsc` issue is fixed in Yarn 0.28. Let's pick this up again.

File: README.md [hunk 7]
```diff
@@ -34,7 +34,7 @@ The other packages (`site-docs` and `site-landing`) are not published to NPM as
 We use [Lerna](https://lernajs.io/) to manage inter-package dependencies in this monorepo.
 Builds are orchestrated via [Gulp](http://gulpjs.com/) tasks.
 
-__Prerequisite__: Node.js v6 or v7
+__Prerequisite__: Node.js v6 or v7, Yarn v0.24+
```
Comment (adidahiya) [inline README.md]: this should now say yarn 0.28+

File: circle.yml [hunk 8]
```diff
@@ -16,27 +16,42 @@ general:
...
   cache_directories:
-    - packages/core/node_modules
-    - packages/datetime/node_modules
-    - packages/docs/node_modules
-    - packages/labs/node_modules
-    - packages/site-docs/node_modules
-    - packages/site-landing/node_modules
-    - packages/table/node_modules
+    - ~/.cache/yarn
+    - ~/blueprint/packages/core/node_modules
```
Comment (adidahiya) [inline circle.yml]: instead of blueprint we can use `$CIRCLE_PROJECT_REPONAME`, like above

**Pull request merged**